app/models/mvi.rb

```diff
@@ -121,18 +121,23 @@ def mvi_response
 
   def response_from_redis_or_service
     do_cached_with(key: @user.uuid) do
-      profile = mvi_service.find_profile(@user)
-      self.class.redis_namespace_ttl = set_ttl(profile)
-      profile
+      mvi_service.find_profile(@user)
     end
   end
 
   def mvi_service
     @service ||= MVI::Service.new
   end
 
-  def set_ttl(profile)
-    if profile.status == MVI::Responses::FindProfileResponse::RESPONSE_STATUS[:ok]
+  def save
+    return false unless valid?
+    redis_namespace.set(attributes[redis_namespace_key], Oj.dump(attributes))
+    expire(record_ttl)
+    @persisted = true
+  end
```

Comment:
I think instead of duplicating the code from `redis_store.save`, I think a cleaner solution probably exists. One example I thought up:

```ruby
def save 
  ...
  expire(record_ttl) if record_ttl
  @persisted = true
end

def record_ttl
  redis_namespace_ttl
end
```

And then mvi.rb would only override `def get_ttl`. Might be able to use `super` somehow as well ¯\\\_(ツ)\_/¯

Reply:
yeah, you're right. How about this?